src/box/tuple_format.c

```diff
@@ -1517,7 +1540,16 @@ tuple_format_apply_defaults(struct tuple_format *format, const char **data,
 		if (is_null)
 			field = tuple_format_field(format, i);
 
-		if (is_null && tuple_field_has_default(field)) {
+		if (is_null && tuple_field_has_default_func(field)) {
+			struct field_default_value *d = &field->default_value;
+			const char *ret_data;
+			uint32_t ret_size;
+			if (d->func->call(d->func, d->data, d->size,
+					  &ret_data, &ret_size) != 0)
+				return -1;
```

Comment:
Nit: Let's add an inline wrapper around `call` and name it `field_default_func_call`. Let's call the actual implementation `field_default_func_call_impl` and add a comment to it explaining why we need it (that is, to avoid linking dependencies).

Reply:
Done.